src/calculated-property-rule.ts

```diff
@@ -102,6 +102,10 @@ export class CalculatedPropertyRule extends Rule {
 
 		// modify list properties to match the calculated value instead of overwriting the property
 		if (this.property.isList) {
+			// calculated lists need to be initialized to prevent infinite recursion when trying to get the current value below
+			if (Property$pendingInit(obj, this.property))
+				Property$init(this.property, obj, []);
```

Comment:
Minor quibble. Is there a way to do this without using the "private" `Property$` functions? There is an `isInited` method on the property that you could use, though maybe the "pending init" concept is important here? And if the property isn't initialized then you could instead set the property value (the no change and update logic wouldn't be necessary). I'm not sure if the setter would do quite what you want in this case (looks like it might also require the property to be initialized), though if not then I think you could argue that's a bug.

Reply:
I was able to use an alternate approach to fixing the issue, which was simply to allow the existing `Property$ensureInited` logic to run for a calculated list property, since it is invalid for a list property to not have an actual array value.